.github/workflows/spectral-publish.yml

```diff
@@ -0,0 +1,41 @@
+name: Publish IPA Validation Framework Package
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - CLOUDP-329998
+    paths:
+      - 'tools/spectral/ipa/**'
+
+jobs:
+  generate-changelog:
+    runs-on: ubuntu-latest
+    steps:
+     - uses: actions/checkout@v4
+     - run: npm install -g conventional-changelog-cli
+     - name: Generate and Commit Changelog
+       run: |
+        npm run gen-ipa-changelog
+        git add tools/spectral/ipa/CHANGELOG.md
+        git commit -m "chore(changelog): Update changelog"
+     - name: Push Changelog
+       uses: ad-m/github-push-action@master
+       with:
+         github_token: ${{ secrets.GITHUB_TOKEN }}
+         branch: main
```

Comment:
Unsure if that will work (we do have branch protection rules)

Reply:
I suggest we get changelog generated in separate PR. 
You need specific filters and standards for changelog to be introduced for it to have any value.